mono/metadata/threads.c

```diff
@@ -6128,7 +6128,7 @@ mono_threads_summarize (MonoContext *ctx, gchar **out, MonoStackHash *hashes)
 				case MONO_SUMMARY_EXPECT: {
 					MOSTLY_ASYNC_SAFE_PRINTF("Timed out, thread did not respond to signal\n");
 
-					MonoNativeThreadId old_val = mono_atomic_cas_ptr ((gpointer) &summarizing_thread, NULL /* set */, GINT_TO_POINTER(tid) /* compare */);
+					MonoNativeThreadId old_val = (MonoNativeThreadId) mono_atomic_cas_ptr ((gpointer) &summarizing_thread, NULL /* set */, GINT_TO_POINTER(tid) /* compare */);
```

Comment:
```
static MONO_ALWAYS_INLINE gint64
mono_atomic_cas_generic (gsize size, void *dest, gint64 exch, gint64 omp)
{
	g_assertf (size == 4 || size == 8, "%u", (guint)size);
	switch (size) {
	case 4:
		return mono_atomic_cas_i32 ((gint32*)dest, (gint32)exch, (gint32)comp);
	case 8:
		return mono_atomic_cas_i64 ((gint64*)dest, (gint64)exch, (gint64)comp);
	}
}


static MONO_ALWAYS_INLINE MonoNativeThreadId
mono_atomic_cas_native_thread_id (MonoNativeThreadId *dest, MonoNativeThreadId exch, MonoNativeThreadId comp)
{
	g_assert (sizeof (MonoNativeThreadId) <= sizeof (gsize));
	return (MonoNativeThreadId)mono_atomic_cas_generic (sizeof (MonoNativeThreadId), dest, (gint64)(gsize)exch, (gint64)(gsize)comp);
}

```

?

Reply:
Can we have this where MonoNativeThreadId is defined instead? This way it could use i32 variant for Windows, and pointer size on Posix, or maybe using AX_COMPILE_CHECK_SIZEOF(pthread_t).

Reply:
Agreed their are other ways. But you get the ideas: zero cost abstraction. Macros ok too.

Reply:
Alright, I'll take a look. What's a preferred way to push updated patches by the way? Should I force update, or add more commits on top that would fix original ones?